site/en/articles/renderingng-fragmentation/index.md

```diff
@@ -0,0 +1,204 @@
+---
+{% Aside %}
+This post is a part of a series on the Chromium rendering engine. Check out the rest of the series to learn more about [RenderingNG](/articles/renderingng/), the [RenderingNG architecture](/blog/renderingng-architecture/), [key data structures](/blog/renderingng-data-structures/), [VideoNG](/blog/videong/), [LayoutNG](/blog/layoutng/)
+and [BlinkNG](/blog/blinkng/).
+{% endAside %}
+
+I'm Morten Stenshorne, a layout engineer on the Blink rendering team at Google. I've been involved in browser engine development since the early 2000s, and I've had a lot of fun, such as helping make the [acid2 test](https://en.wikipedia.org/wiki/Acid2#Timeline_of_passing_applications) pass in the Presto engine (Opera 12 and earlier), and reverse-engineering other browsers to fix table layout in Presto. I've also spent more of those years than I'd like to admit on block fragmentation, and, in particular, [multicol](https://www.w3.org/TR/css-multicol-1/) in Presto, WebKit, and Blink. During the past few years at Google I have mainly been focusing on leading the work of adding block fragmentation support to [LayoutNG](/articles/layoutng/). Join me in this deep-dive into the block fragmentation implementation, as it may very well be the last time that I implement block fragmentation. :)
+
+## What is block fragmentation?
+
+[Block fragmentation](https://www.w3.org/TR/css-break-3/) is about splitting a CSS block-level box (such as a section or paragraph) into multiple fragments when it doesn't fit as a whole inside one fragment container (called a [_fragmentainer_](https://www.w3.org/TR/css-break-3/#fragmentainer). A fragmentainer is not an element, but represents a column in multi-column layout, or a page in paged media. For fragmentation to happen, the content needs to be inside a fragmentation context. A fragmentation context is most commonly established by a multi-column container (content will be split into columns), or when printing (content will be split into pages). A long paragraph with many lines may need to be split into multiple fragments, so that the first lines are placed in the first fragment, and the remaining lines are placed in subsequent fragments. 
+
+<figure>
+{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/cCaTuW0K4aGvX1HHSH6d.png", alt="A paragraph of text broken into two columns.", width="800", height="576" %}
+
+<figcaption>In this example a paragraph has been split into two columns using multi-column layout. Each column is a fragmentainer, representing a fragment of the fragmented flow.</figcaption>
+</figure>
+
+
+Block fragmentation is analogous to another well-known type of fragmentation: line fragmentation. Any inline element that consists of more than one word (any text node, any `<a>` element, and so on), and allows line breaks, may be split into multiple fragments. Each fragment is placed into a different line box. A line box is the inline fragmentation equivalent to a _fragmentainer_ for columns and pages.
```

Comment:
```suggestion
Block fragmentation is analogous to another well-known type of fragmentation: line fragmentation (otherwise known as "line breaking"). Any inline element that consists of more than one word (any text node, any `<a>` element, and so on), and allows line breaks, may be split into multiple fragments. Each fragment is placed into a different line box. A line box is the inline fragmentation equivalent to a _fragmentainer_ for columns and pages.
```